core/src/main/python/wlsdeploy/util/target_configuration.py

```diff
@@ -101,6 +109,21 @@ def uses_credential_secrets(self):
         """
         return self.get_credentials_method() in [SECRETS_METHOD, CONFIG_OVERRIDES_SECRETS_METHOD]
 
+    def generate_script_for_secrets(self):
+        """
+        Determine if it needs to generate shell script for creating secrets.
+        :return: True if it is not equal to json
+        """
+        return not self.get_credentials_output_method() in ['json']
+
+
+    def generate_json_for_secrets(self):
+        """
+        Determine if it needs to generate json file for creating secrets.
+        :return: True if generating json file, False otherwise
+        """
+        return self.get_credentials_output_method() in ['json']
```

Comment:
will this call return false if self.get_credentials_output_method() returns None? I want to be sure it works if credentials_output_method is not in the target JSON. Someone could have an existing target file they don't know to update.

Reply:
It should,  get_credentials_output_method() uses dictionary_util.get_element() which returns None and None in ['json'] returns false.